internal/internal_workflow_testsuite.go

```diff
@@ -2483,9 +2533,66 @@ func (env *testWorkflowEnvironmentImpl) RequestCancelNexusOperation(seq int64) {
 	}
 }
 
+func (env *testWorkflowEnvironmentImpl) RegisterNexusAsyncOperationCompletion(
+	service string,
+	operation string,
+	operationID string,
```

Comment:
I wonder if this would be intuitive for users to specify the operationID at registration time.
Also don't we need locks to protect the map here?

Reply:
> I wonder if this would be intuitive for users to specify the operationID at registration time.
The user needs to specify the operation id when specifying the return of the mock call.

> Also don't we need locks to protect the map here?
Why do we need to protect this map?

Reply:
I can see users registering a completion handler while the test env is running so I would like to avoid data races in those cases.